src/v/cloud_storage/remote_partition.cc

```diff
@@ -457,49 +457,19 @@ class partition_record_batch_reader_impl final
             query = model::offset_cast(config.start_offset);
         }
         // Find manifest that contains requested offset or timestamp
-        int retry_quota = 4;
-        while (retry_quota-- > 0) {
-            auto stm_start_offset
-              = _partition->_manifest_view->stm_manifest().get_start_offset();
-            if (!stm_start_offset.has_value()) {
-                vlog(
-                  _ctxlog.error, "STM manifest has no start offset; no data");
-                co_return;
-            }
-            auto cur = co_await _partition->_manifest_view->get_cursor(query);
-            if (cur.has_failure()) {
-                vlog(
-                  _ctxlog.error,
-                  "Failed to query spillover manifests: {}, query: {}",
-                  cur.error(),
-                  query);
-                co_return;
-            }
-            auto new_stm_start_offset
-              = _partition->_manifest_view->stm_manifest().get_start_offset();
-            if (likely(new_stm_start_offset == stm_start_offset.value())) {
-                _view_cursor = std::move(cur.value());
-                if (
-                  _view_cursor->manifest()->get().get_start_offset().value()
-                  == new_stm_start_offset) {
-                    _initial_stm_start_offset = new_stm_start_offset;
-                }
-                initialize_reader_state(
-                  _view_cursor->manifest().value(), config);
-                co_return;
-            }
+        auto cur = co_await _partition->_manifest_view->get_cursor(query);
+        if (cur.has_failure()) {
             vlog(
-              _ctxlog.info,
-              "STM manifest start offset moved from {} to {} due to spillover "
-              "or truncation; retrying lookup {}",
-              stm_start_offset,
-              new_stm_start_offset,
+              _ctxlog.error,
+              "Failed to query spillover manifests: {}, query: {}",
+              cur.error(),
               query);
+            co_return;
         }
-        vlog(
-          _ctxlog.warn,
-          "Couldn't initialize cursor for {} after retrying",
-          query);
+        _view_cursor = std::move(cur.value());
+        co_await _view_cursor->sync_manifest();
+        initialize_reader_state(_view_cursor->manifest().value(), config);
```

Comment:
Seastar question: what gives us the guarantee that after the sync completes, before control returns to L471, that the manifest hasn't changed? Is it guaranteed by the fact that `seek()` has no scheduling points after setting `_current`?

Reply:
Isn't this still a problem if `sync_manifest()` *didn't* do anything to fix the state, and then `initialize_reader_state()`? For instance, regardless of if `sync_manifest()` re-seeked, can't a spillover happen in "between" L470 and 471 by virtue of there being a scheduling point then?

Reply:
that's a good point

Reply:
fixed this